### PR TITLE
[move-lang][error migration] Migrate parser/ errors

### PR DESCRIPTION
**Pull request by tnowacki**
- Migrated all errors in parser/ to new diagnostic system

## Motivation

- Had to cleanup/update the API a bit more. I had them as separate commits but had issue rebasing so needed to squash, sorry

## Test Plan

- Ran the tests

### CODE REVIEW

File: language/move-lang/src/errors/diagnostic_codes.rs
Comment (modocache) [inline language/move-lang/src/errors/diagnostic_codes.rs:178]: Maybe it's just me, but I found it odd that the `MAX` member of the `Severity` type is `Warning`. It made me do a double-take because wouldn't the "maximum" severity be a blocking error?

Looking below at how this member is used for the "default" severity, I wonder if it ought to be renamed to something besides `MAX`? Or, more likely, I'm misunderstanding something about this code... 😅

File: language/move-lang/src/parser/mod.rs
Comment (modocache) [inline language/move-lang/src/parser/mod.rs:63]: Is this a TODO for you before landing this, or for someone else to read later? What prevents warnings from appearing here?

**Review by modocache — DISMISSED** [review]
I had a question and a nitpick about TODOs but other than that this looks good! 🚀 

(For my own benefit just noting that this is to continue the work from https://github.com/diem/diem/commit/b1de5f8ae31b5f640d8eb97a6587b5ffcf810796, to update this based on the codespan crate's new behavior.)

File: language/move-lang/src/parser/mod.rs
Comment (tnowacki) [reply language/move-lang/src/parser/mod.rs:63]: It's intentional. The parser pass is a bit funky and out of alignment with the code pattern of the other passes. It would take some refactoring to fix that to allow the non-blocking errors to go through, so that will be for a future PR

File: language/move-lang/src/errors/diagnostic_codes.rs
Comment (tnowacki) [reply language/move-lang/src/errors/diagnostic_codes.rs:178]: I went with `MAX` in the orderable sense, that is for any `x: Severity` `x < Severity::MAX`. I tried to match the naming for `u64` and the like, but it definitely is a bit confusing with the word `Severity` as those tend to have their own meaning. 

Maybe I should flip all of these orderings? Might make more sense to have the *least severe* diagnostic type be ordered the least. Any thoughts?

Comment (tnowacki) [reply language/move-lang/src/errors/diagnostic_codes.rs:178]: I'm actually so convinced that this is the right switch that I'm going to do that. Hopefully I don't regret not waiting for a response, lol

Comment (modocache) [reply language/move-lang/src/errors/diagnostic_codes.rs]: Nice! I think that makes sense. (Sorry, I'm still trying to convince GitHub to send me push notifications...)

**Review by modocache — APPROVED** [review]

File: language/move-lang/src/errors/diagnostic_codes.rs
Comment (tnowacki) [reply language/move-lang/src/errors/diagnostic_codes.rs]: No need to be sorry, lol. I sent that message super late your time and responded to it soon after. But glad it makes sense now ;) ... hopefully

File: language/move-lang/src/parser/syntax.rs
Comment (bob-wilson) [inline language/move-lang/src/parser/syntax.rs:1790]: This indentation is strange. Is this what `cargo fmt` does with it?

**Review by bob-wilson — APPROVED** [review]
This looks good. I like the new message formats from the updated codespan, especially the way that it shows the lines of code in order. I wasn't sure how that would work, but now that I see it, I really like it.

I'm approving but could you comment on your plans for the `_deprecated` functions here? I assume that's just a temporary step, right?

**Comment by tnowacki** [comment]
> This looks good. I like the new message formats from the updated codespan, especially the way that it shows the lines of code in order. I wasn't sure how that would work, but now that I see it, I really like it.
> 
> I'm approving but could you comment on your plans for the `_deprecated` functions here? I assume that's just a temporary step, right?

I'll fix any formatting issues in a future PR (one will be up as soon as this lands)

The `_deprecated` stuff will die as soon as all of these migrations land. Just missed some on the initial PR

**Pull request merged**
